sdk/core/azure-core/azure/core/pipeline/policies/__init__.py

```diff
@@ -25,7 +25,12 @@
 # --------------------------------------------------------------------------
 
 from ._base import HTTPPolicy, SansIOHTTPPolicy, RequestHistory
-from ._authentication import BearerTokenCredentialPolicy, AzureKeyCredentialPolicy, AzureSasCredentialPolicy
+from ._authentication import (
+    AzureKeyCredentialPolicy,
+    AzureSasCredentialPolicy,
+    BearerTokenCredentialPolicy,
+    MultitenantTokenCredentialPolicy,
```

Comment:
AAD docs use "Multitenant", but there was (at one point) an option in .NET's Identity library for [AllowMultiTenantAuthentication](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/CHANGELOG.md#breaking-changes-from-150-beta4). This never got released in a stable version though, so I don't know if there was any architect input on the naming.

Technically, .NET's policy name for this feature is [TableBearerTokenChallengeAuthorizationPolicy](https://github.com/Azure/azure-sdk-for-net/blob/1722fc8329bd848d400faf73daeba06d9d718cc1/sdk/tables/Azure.Data.Tables/src/TableBearerTokenChallengeAuthorizationPolicy.cs#L28) -- obviously that's Tables-specific, but it does suggest the idea of something like `BearerTokenChallengePolicy`. I don't think that's a bad name either, and honestly, I think using "Challenge" instead of "Multitenant" makes sense. The policy is specifically about handling auth challenges, and can be used for more (or less than) cross-tenant auth.

Reply:
I would be okay with something like BeaerTokenChallenge[Credential]Policy.
I agree I think it makes more sense than Multitenant - as that's more of an outcome rather than a specific part of the design.